### PR TITLE
PR: feature/animated-screens

### PR DESCRIPTION
**Pull request by tombu**
This PR introduces the ability to animate the screens. The new mixin `Animatable` carries this ability. It hooks into the `didInsertElement` method of the respective screen view (for intro-animation) and into the `destroy` method of the screen.

You can define animations by defining `animateIn` or rather `animateOut`. To create a valid animation you have to use the `animate()` method of the mixin. This method takes three parameters: 
- options (object containing `duration` and/or `delay`)
- animation function (any jQuery animation thingi)
- callback function (not required)

Things I still would like to implement:
- block touch/mouse events while animation is running (should we?)
- automatically chaining animation (this would allow to remove the 'duration' and 'delay' options)
- cancel animation (not sure if we need that - we may need it, if we don´t block user interaction while the animation is running)

What do you think about the missing steps?

### CODE REVIEW

File: app/assets/javascripts/slotcars/build/views/build_screen_view.js.coffee [hunk 1]
```diff
@@ -13,6 +13,8 @@ Build.BuildScreenView = Ember.View.extend
     @set 'editButton', Shared.Button.create()
     @set 'publishButton', Shared.Button.create()
 
+  didInsertElement: -> # Don´t use - it is overridden by Shared.Animatable
```
Comment (DominikGuzei) [inline app/assets/javascripts/slotcars/build/views/build_screen_view.js.coffee]: why do we need it here anyway if it is just an empty method?

File: app/assets/javascripts/slotcars/shared/mixins/animatable.js.coffee [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Shared.Animatable = Ember.Mixin.create
+
+  animation: null
+  view: Ember.required()
+
+  append: ->
+    # overrides `didInsertElement` of the view to plug-in the animation
+    @view.didInsertElement = => @animateIn() if @animateIn
```
Comment (DominikGuzei) [inline app/assets/javascripts/slotcars/shared/mixins/animatable.js.coffee]: you could create a decorated method here that also calls the original `didInsertElement` method with all params! And thus remove all the "don't use this " warnings above

**Comment by DominikGuzei** [comment]
I always see a short flash of content at the end of the transitions! you have any idea?
e.g: when I go from home -> build I see the home screen flashing in for a few milliseconds after the fade in of the build screen

**Comment by stravid** [comment]
I think @tombu couldn't find a way to fix this. As far as I can remember there are various people on the internetz with this problem..

**Comment by DominikGuzei** [comment]
is it ember.js related?

**Comment by stravid** [comment]
I don't think so. It only happens on a screen which uses transitions.

**Comment by DominikGuzei** [comment]
hmm... it kind of destroys the positive aspects of the whole idea of transitions. For me it's really annoying to see that flickering everytime, so the transition doesn't add a real benefit :-(

**Comment by tombu** [comment]
I found a solution here: http://stackoverflow.com/questions/2975217/how-to-fix-flicker-when-using-webkit-transforms-transitions

**Pull request merged**